ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,279 @@
+
+### Component Architecture
+
+| Component | Role | Optional |
+|-----------|------|----------|
+| VS Code Plugin | Main interface for onboarding and configuration | No |
+| Git Repos | Source of truth for Konflux config | No |
+| GitOps Registration Service | API for automated registration of GitOps repos as tenants | **Yes** (recommended) |
+| ArgoCD Service | Continuous GitOps sync and deployment | **Yes** (recommended) |
+| Konflux UI | Monitoring, runtime insights, and operational actions (non-configuration) | No |
+
+### VS Code Plugin Design
+
+**Core Features:**
+- Forms and wizards to scaffold and edit YAML objects
+- Built-in schema-aware YAML editor
+- Linting and validation against Konflux schemas
+- Git-aware diff view and PR-ready commit generation
+- Optional: preview pipeline graph / dependency resolution
+
+
+**GitOps Flow:**
+
+1. Developer creates an empty git repo with Github, Gitlab, etc.
+4. **If first time**: Uses wizard to register repo with Tenant Registration Service (declares namespace name, configures ArgoCD)
+5. Developer is prompted by UI to install / open repo in VS Code
+3. Launches Konflux plugin and uses it to configure new components, tests, releasePlans
+6. Plugin generates YAML objects in repo
+7. Developer commits and pushes code (optionally via PR, with CI,reviews etc.)
+9. ArgoCD automatically syncs merged changes to the namespace
+
+### Multi-Repo GitOps Model & Optional Services
+
+**GitOps Registration Service (Optional):**
+Konflux will optionally provide a **GitOps Registration Service** API that automates the registration and lifecycle of GitOps repositories as first-class tenants. This service is recommended for production deployments but not required for basic Konflux functionality.
+
+**ArgoCD Service (Optional):**
+ArgoCD can be optionally deployed as a continuous GitOps synchronization service. While recommended for production use, Konflux can operate without ArgoCD, though this results in reduced automation and requires manual synchronization of GitOps repositories.
+
+**Impact of Optional Services:**
+- **With both services**: Full automated GitOps experience with continuous deployment and streamlined onboarding
+- **Without both services**: Konflux remains fully functional but requires manual GitOps repository setup and synchronization. This configuration provides:
+  - Smaller deployment footprint suitable for lower-spec machines
+  - Simplified setup for testing and development environments  
+  - Reduced operational complexity for teams preferring manual control
+  - Degraded user experience with manual setup steps
+  - Reduced disaster recovery capabilities due to lack of automated drift detection
+
+**Tenant Model:**
+- **Tenant Definition**: A tenant is equivalent to a Kubernetes namespace
+- **1:1 Mapping**: Each GitOps repository maps to exactly one Kubernetes namespace
```

Comment:
Why this limitation is needed?

Reply:
I like it, its clean and easy to understand / enforce and we have long said that namespaces are the logical team boundaries for Konflux. It fits our model.